source/zipkin/span_buffer.cc

```diff
@@ -0,0 +1,40 @@
+#include "zipkin/span_buffer.h"
+
+namespace Zipkin {
+
+void SpanBuffer::allocateBuffer(uint64_t size) {
+  span_buffer_.resize(size);
+  size_ = size;
+}
+
+bool SpanBuffer::addSpan(Span&& span) {
+  if (next_position_ == size_) {
+    // Buffer full
+    return false;
+  }
+  // span_buffer_[next_position_++] = &span;
+  span_buffer_[next_position_++] = std::make_shared<Span>(span);
+
+  return true;
+}
+
+void SpanBuffer::flush() {
+  next_position_ = 0;
+  std::fill(span_buffer_.begin(), span_buffer_.end(), nullptr);
+}
+
+std::string SpanBuffer::toStringifiedJsonArray() {
+  std::string stringifiedJsonArray = "[";
```

Comment:
All variables should use snake case. I won't comment on this further but please fix everywhere.